src/control/server/config/server.go

```diff
@@ -391,7 +390,7 @@ func getAccessPointAddrWithPort(log logging.Logger, addr string, portDefault int
 }
 
 // Validate asserts that config meets minimum requirements.
-func (cfg *Server) Validate(ctx context.Context, log logging.Logger) (err error) {
+func (cfg *Server) Validate(log logging.Logger, hugePageSize int, fis *hardware.FabricInterfaceSet) (err error) {
```

Comment:
Hmm, I had the idea (from a long while ago) that we used these server and engine config `Validate()` functions as a sort of bare minimum check that all the required parameters are there, and correctly formatted. I.e. not necessarily pulling in all these external pieces of data. I don't know how essential that is at this stage, though. It was the motivation for keeping a separate `ValidateFabric()` method for the engine config though.

Reply:
It makes sense to me to do this all in one place and not in a number of separate functions, to me that makes things more confusing for someone trying to follow the code. hugePageSize is required to validate the nrHugePages in config and fis is passed into engine Validate() that is also now consolidated into one function. The other changes to this function involve improving the logical flow and moving things around.